app/controllers/business_entities_controller.rb

```diff
@@ -0,0 +1,110 @@
+class BusinessEntitiesController < ApplicationController
+  def index
+    authorize :business_entity
+    list_jurisdictions
+  end
+
+  def new
+    redirect_to office_business_entities_path if valid_business_entity || !jurisdiction
+    @business_entity = BusinessEntity.new
+    authorize @business_entity
+  end
+
+  def create
+    @business_entity = business_entity_service.build(business_entity_params)
+    authorize @business_entity
+    if @business_entity.save
+      redirect_to office_business_entities_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    authorize business_entity
+  end
+
+  def update
+    bes = business_entity_service
+    new_be = bes.check_update(business_entity_params)
+    authorize new_be
+
+    if bes.persist_update!(new_be)
+      redirect_to office_business_entities_path
```

Comment:
> I see why @colinbruce is using abbreviations, I use them sometimes because of the line length restriction, which I find annoying at times.

True, they are annoying. But I feel that those are the constraints placed in order to enforce better discipline when developing. There are always ways to work around the line length, without sacrificing readability.